ash-bergs/pantry-pal
# feat: add lists

- Adds `lists` table to db
- Adds `itemLists` join table for many to many relationship between lists and items 
- Creates add list form - still needs some style for the form itself 
- Adds functionality for the add list form with `addListForm.ts`
- Creates new modal for the add list form to live in 
- Adds modal functionality with `addListModal` 
- Adds `addList` method to the `ItemManager` class 
- Updates some button styles, relying on class names instead of ids, because that makes more sense and makes the styles more shareable
- Adds the "Add List" right by the "Add Item" button 

<img width="391" alt="image" src="https://github.com/user-attachments/assets/d3c904fe-4af8-456e-a02b-daa52ccb8638">

## Testing 
1. `npm run build`
2. `npm run dev`
3. Open inspector -> Application -> Indexdb - you should see the `lists` and `itemLists` tables in the indexdb
4. Click add list button
5. Should not be able to add list without giving it a name 
6. Adding a list should update the `lists` table in local indexdb
